src/Collection/CollectionTrait.php

```diff
@@ -470,6 +470,9 @@ public function toArray($preserveKeys = true)
             $items = $iterator->getArrayCopy();
             return $preserveKeys ? $items : array_values($items);
         }
+        if ($preserveKeys && get_class($iterator) === 'RecursiveIteratorIterator') {
```

Comment:
That it emits the same array keys for each of the tree levels, making the array conversion lose values